### ISSUE TITLE
Guest user access.

### ISSUE THREAD

**Issue by jordy2254** [photoview/photoview#930]
**Is your feature request related to a problem? Please describe.**
The ability to be able to access a photo view with a guest account that doesn't require login. Per PR #861 

**Describe the solution you'd like**
The ability to configure a guest account to access photoview that doesn't require any login

> Discussion required. This will require input from the community to work out how to shape this for the future of photoview.

**Comment by pcmike** [comment]
Just posting to say the functionality mentioned is what I'm currently searching for and that's how I came across photoview. I like the core functionality, but in reality I just want a simple way to present photos on a domain for everyone to see (so long as I've marked the album publicly viewable). I don't want anyone to have to log in to anything.

**Comment by jordy2254** [comment]
@pcmike 
That's great. Could you suggest from your perspective how you'd like this to work?
- what would be the flow for a user to get guest access
- what would be the flow to login as a user when using guest

Please thumbs up the original post for the issue too so we can keep track of how many people want this.

**Comment by pcmike** [comment]
The way this would work for me would be you'd have photos.tld and when a visitor visits the domain they would just see a bunch of albums/photos/etc. They would be able to see the entire installation, so long as the albums are marked public. There would be no logging in to see and interact with things marked public. Logging in would be reserved for the admin or an actual user who has photos on the installation and when logging in they can do anything, most important of which would be choosing what's marked public or not.

**Comment by kkovaletp** [comment]
I'd rephrase that (please correct me if I'm wrong and suggest changes to this design):
- a new property for a media item should be introduced: `isPublic = <true | false>` - false by default
- a new checkbox or boolean slider should be added to the Media info sidebar in the Sharing options section to control the "isPublic" property, as well as to the Album settings sidebar in the same section (changing it from an album sets the new value to all media inside that album independently of what was the previous value of those media items). This checkbox or slider should have the (i) pop-up explaining that the setting takes effect only when Guest access is enabled by Admin.
- A login button should be added to the main page header on the left side of the search field. It should open our current Login page. For logged-in users, it should be replaced by the Logout button, which now is located in Settings
- a new Admin setting should be added to the User management section: Enable Guest access <enabled | disabled> - disabled by default. It would be great to also have a short description right under this setting on the page.

- when the Guest access feature is turned on in the Admin settings page:
1. A confirmation pop-up appears with a warning that the user management model is going to change and not auth'ed users will get access to the media, marked as public
2. Landing page is switched from the Login to the main page
3. The main page shows for guests only Login, Search, Timeline, Albums, and Places (under the question, as geolocation is more sensitive private info, which users might not want to share even if corresponding media is marked as public). People and Settings should be hidden and not accessible from API
4. If no media is marked as public, there should be a placeholder for guests, telling that no media is shared yet on the server
5. Search should return only media, which are accessible by current user: guests should get search results only with public media items
6. Favorites functionality should be hidden and disabled for guests

- when the Guest access feature is turned off on the Admin settings page:
1. A confirmation pop-up appears with a warning that the user management model is going to change and not auth'ed users will lose access to the media, marked as public
2. Landing page is switched from the main to the Login page
3. The properties' values for existing media are not affected by enabling or disabling the feature, so the next time it is enabled, all previously marked as public media will be shown to guests again

**Comment by jordy2254** [comment]
@kkovaletp implementation details should be left down to the contributer doing the implementation, we only really want to cover the flow and intended use cases here.

@pcmike 
Sounds great, slightly different from what I was thinking but suites a different use case. My desired use case as an admin user would be to assign specific albums and not give user's the ability. (The same as creating a new user), Although I love your use case too so I think we should cater for both.

How would you expect to interact with the system when using guest access? Would you expect to see the login screen as now and then have a 'continue as guest' button or would you expect direct access with a login button available on the menu bar? Again maybe we could do both if there is enough demand...

**Comment by pcmike** [comment]
I would expect no continue as guest.. just directly displaying the albums/photos and then if an admin/user wants to interact with the installation then they can just click a simple icon somewhere that brings up a login screen. The goal should be for it to just look like a photo wall/gallery from a random visitor's perspective.

I must say.. I've been trying to get photoview to work this morning via docker and up till now I've been unsuccessful. I originally tried with having both the mariadb service with a host mounted volume for the db and a host mounted volume for the cache, no dice. I then switched it use sqlite driver and getting rid of the mariadb service and again a host mounted volume for db and cache, still no dice. Not sure what else to try. I've detailed everything on discord, but I can always create a new issue here if needed.

Thank you for even giving my opinion a voice, considering I haven't even used photoview yet. :)

**Comment by uxigene** [comment]
First of all, thanks for this awesome photo library. In addition I would add that guests should not be able to label faces.

**Comment by tschaerni** [comment]
When I did setup photoviewer just before, I expected that there would be public viewable albums, but apart from creating a sharing link, there is no such feature.
Sadly that makes photoviewer unusable for my usecase. I really like how simple and clean it is, but without public access I have to switch to a different project :/

I hope to keep getting notified if that features will be implemented ;)

**Comment by kkovaletp** [comment]
The recent question in our Discord triggered the idea of a workaround for this feature request until it is not implemented, however, it requires direct access to the DB and making a change, which might be dangerous in case of a mistake, as potentially might corrupt data integrity and lead to data loss or Photoview instance corruption.

Do it only if you know what are you doing.

- Go to the UI and create a public share for an album (any one)
- Go to the DB, find the `share_tokens` table, and modify the last row with just created share: replace the `album_id` with `1`
- Open the previously created link in the incognito browser window (to make sure that the session from your main browser is not reused)

Of course, this workaround provides only limited functionality and doesn't give access to many features, but at least your guests can view the gallery

**Comment by amirasyraf** [comment]
Would love this. Or a way to just disable auth.

I just want a good gallery which kinda looks like Google Photos to showcase all my photos

**Comment by tmanok** [comment]
Just a follow up, this still seems like a great feature to add, in addition to the existing shareable links, have a public section. It brings up permissions more broadly. Currently, the setup seems to be each user has a home folder and can see everything in their home folder- in order for users to see things in each other's home folders, the photos need to be copied. This isn't very practical, although perhaps I should experiment with symlinks later. 

In any case, a public version of the site and private user with folders visible to multiple users would be really welcome. 
Thanks! Such an awesome project, it's been so impactful on my life and workflow. 

Tmanok

**Comment by kkovaletp** [comment]
@tmanok, we are still waiting for a contributor willing to take this feature for implementation. 
The sharing between users use case has its own ticket #608
